src/test/java/uk/gov/dwp/dataworks/provider/KMSDataKeyGeneratorProviderTest.java

```diff
@@ -1,24 +1,157 @@
 package uk.gov.dwp.dataworks.provider;
 
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.model.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
 import uk.gov.dwp.dataworks.dto.GenerateDataKeyResponse;
-import org.springframework.util.Assert;
+import uk.gov.dwp.dataworks.errors.CurrentKeyIdException;
+import uk.gov.dwp.dataworks.errors.DataKeyGenerationException;
 
+import java.nio.ByteBuffer;
+import java.util.Base64;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.BDDMockito.given;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest()
+@ActiveProfiles({"UnitTest", "KMS"})
 public class KMSDataKeyGeneratorProviderTest {
 
-    //@Test
-    public void canGenerateDataKey() {
-        KMSCurrentKeyIdProvider keyIdProvider = new KMSCurrentKeyIdProvider();
-        KMSDataKeyGeneratorProvider providerUnderTest = new KMSDataKeyGeneratorProvider();
+    @Test
+    public void generateDataKey() {
+        String dataKeyEncryptionKeyId = "DATAKEYENCRYPTIONKEYID";
+        String plainTextKey = "PLAINTEXTKEY";
+        String encryptedDataKey = "ENCRYPTEDDATAKEY";
+        GenerateDataKeyResult result = new GenerateDataKeyResult();
+        result.setKeyId(dataKeyEncryptionKeyId);
+        result.setPlaintext(ByteBuffer.wrap(plainTextKey.getBytes()));
+        result.setCiphertextBlob(ByteBuffer.wrap(encryptedDataKey.getBytes()));
+        GenerateDataKeyRequest generateDataKeyRequest = new GenerateDataKeyRequest();
```

Comment:
Feels a bit words to need to use setters on a DTO. Are you sure you don't want to set them in the constructor? The request is not valid without both being set